source/DSCResources/MSFT_SccmIniFIle/MSFT_SccmIniFIle.psm1

```diff
@@ -0,0 +1,1107 @@
+        [String]
+        $DeleteClientLanguages,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $SQLServerName,
+
+        [Parameter(Mandatory = $true)]
+        [String]
+        $DatabaseName,
+
+        [Parameter()]
+        [String]
+        $SqlSsbPort,
+
+        [Parameter()]
+        [String]
+        $SQLDataFilePath,
+
+        [Parameter()]
+        [String]
+        $SQLLogFilePath,
+
+        [Parameter()]
+        [String]
+        $CCARSiteServer,
+
+        [Parameter()]
+        [String]
+        $CasRetryInterval,
+
+        [Parameter()]
+        [ValidateRange(0, 100)]
+        [uint16]
+        $WaitForCasTimeout,
+
+        [Parameter(Mandatory = $true)]
+        [Boolean]
+        $CloudConnector,
+
+        [Parameter()]
+        [String]
+        $CloudConnectorServer,
+
+        [Parameter()]
+        [Boolean]
+        $UseProxy,
+
+        [Parameter()]
+        [String]
+        $ProxyName,
+
+        [Parameter()]
+        [String]
+        $ProxyPort,
+
+        [Parameter()]
+        [Boolean]
+        $SAActive,
+
+        [Parameter()]
+        [Boolean]
+        $CurrentBranch
+    )
+
+    $IniFilePath = $IniFilePath.TrimEnd('\')
+    Write-Verbose "Getting file content of $IniFilePath\$IniFileName"
+    $iniContent = Get-Content -Path "$IniFilePath\$IniFileName" -ErrorAction SilentlyContinue
+
+    $systemParameters = @('Verbose','Debug','ErrorAction','WarningAction','InformationAction','ErrorVariable','WarningVariable','InformationVariable','OutVariable','OutBuffer','PipelineVariable')
+    $Testparameters = (Get-Command -Name 'Get-TargetResource').Parameters.values | Select-Object -Property  Name,ParameterType
+
+    if ($iniContent)
+    {
+        $iniParameters = @{}
+        foreach ($line in $iniContent)
+        {
+            if ($line -match '=')
+            {
+                $iniParameters += @{
+                    (($line.split('=')[0]).Trim(' ')) = (($line.split('=')[1]).Trim(' '))
+                }
+            }
+        }
+
+        $getParameters = @{}
+        foreach ($param in $testParameters)
+        {
+            if ($systemParameters -notcontains $param.Name)
+            {
+                if ($($iniParameters.$($param.Name)))
+                {
+                    Write-Verbose -Message "Adding: $($param.Name) - $($iniParameters.$($param.Name))"
+                    $getParameters.Add($($param.Name),$($iniParameters.$($param.Name)))
+                }
+                else
+                {
+                    Write-Verbose -Message "Adding: $($param.Name) - $null"
+                    $getParameters.Add($($param.Name),$null)
+                }
+            }
+        }
+    }
+    else
+    {
+        Write-Verbose "Could not find $IniFilePath\$IniFileName. "
+        Write-Verbose 'Results will contain parameters passed to configuration.'
```

Comment:
just returning what was inputed just seems weird for get.  would be better to maybe just return the param.name = ''